ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -288,10 +314,56 @@ initFromSnapshot tracer hasFS decLedger decHash cfg streamAPI ss = do
           initStartingWith
             tracer
             cfg
+            readLedgerDb
             streamAPI
             (ledgerDbWithAnchor initSS)
         return (tip, initDB, replayed)
 
+
+mkOnDiskLedgerStDb :: SomeHasFS m -> m (OnDiskLedgerStDb m l)
+mkOnDiskLedgerStDb = undefined
+  -- \(SomeHasFS fs) -> do
+  --   dbhandle <- hOpen fs "ledgerStateDb"
+  --   ...
+
+  --   return OnDiskLedgerStDb
+  --   { ...
+  --     , readKeySets = Snapshots.readDb dbhandle
+
+  --     }
+
+-- | On disk ledger state API.
+--
+--
+data OnDiskLedgerStDb m l =
```

Comment:
We might want to use a type-class here.

Reply:
If we'd use a type-class we would need to pass the handle instead of a value of `OnDiskLedgerStDb`.